ext/SerializationExt.jl

```diff
@@ -0,0 +1,82 @@
+module SerializationExt
+using LRUCache
+using Serialization
+
+# Serialization of large LRUs causes a stack overflow error, so we 
+# create a custom serializer that represents LinkedNodes as Ints
+function Serialization.serialize(s::AbstractSerializer, lru::LRU{K, V}) where {K, V}
+    # Create a mapping from memory address to id
+    node_map = IdDict{LRUCache.LinkedNode{K}, Int}()
+    # Create mapping for first node
+    id = 1
+    first_node = node = lru.keyset.first
+    node_map[node] = id
+    # Go through the rest of the nodes in the cycle and create a mapping
+    node = node.next
+    while node != first_node
+        id += 1
+        node_map[node] = id
+        node = node.next
+    end
+    @assert id == length(lru) == lru.keyset.length == length(lru.dict)
```

Comment:
In principle this can be condensed by using the iterator interface that is defined for `CyclicOrderedSet`. It should produce the exact same code, but reads a little easier.
```suggestion
    for (id, node) in enumerate(lru.keyset)
        node_map[node] = id
    end
```

Reply:
Thanks! Great suggestion.